src/node.cc

```diff
@@ -4747,9 +4754,17 @@ inline int Start(Isolate* isolate, IsolateData* isolate_data,
     do {
       uv_run(env.event_loop(), UV_RUN_DEFAULT);
 
+      {
+        InternalCallbackScope cb_scope(&env, Local<Object>(), { 0, 0 }, true);
```

Comment:
What exactly does this scope provide? Should it also be added to the places in `node_platform.cc` that invoke foreground tasks?

Reply:
> What exactly does this scope provide?

Calling async_hooks hooks + running the microtask queue and/or nextTicks

> Should it also be added to the places in `node_platform.cc` that invoke foreground tasks? 

Hm – I think so, yes. Thanks for pointing that out.

Reply:
FWIW, it would be **exceedingly** helpful if we had actual documentation for how and when things like `InternalCallbackScope` are to be used.

Reply:
@jasnell I agree

@matthewloring I will add this to the other places but that gets a bit tricky, because the platform only sees `Isolate`s but looking up the proper `Environment` for a task is not trivial.